### PR TITLE
feat: [PL-29733]: Handling data deletion for churned accounts.

### PR DESCRIPTION
**Pull request by boopesh007**
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40372)

### CODE REVIEW

**Review by brettzu — COMMENTED** [review]
Are there already unit tests or integration tests that test deletion behavior? I don't see any new tests in this PR.

> Reviewed 6 of 11 files at r1, 7 of 7 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40372#-:-NHQANmz8QSlX_ksWlfI:b7pki40)* status: 13 file(s) to review. 0 of 2 approvals obtained.

**Comment by boopesh007** [comment]
I will add integration tests for each of the collection that am handling @brettzu

**Comment by boopesh007** [comment]
@brettzu I have added the unit tests, can you review now.

**Review by brettzu — APPROVED** [review]
> Reviewed 9 of 9 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40372)* status: 22 file(s) to review. 1 of 2 approvals obtained.

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 5 of 11 files at r1, 7 of 7 files at r2, 8 of 9 files at r3, 1 of 1 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40372)* status: 1 file(s) to review. 1 of 2 approvals obtained.

File: 400-rest/src/main/java/io/harness/perpetualtask/PerpetualTaskScheduleServiceImpl.java [hunk 1]
```diff
@@ -112,4 +114,10 @@ private Query<PerpetualTaskScheduleConfig> queryToGetRecordByAccountIdAndPerpetu
         .field(PerpetualTaskScheduleConfigKeys.perpetualTaskType)
         .equal(perpetualTaskType);
   }
+
+  @Override
+  public void deleteByAccountId(String accountId) {
```
Comment (abhinav-harness) [inline 400-rest/src/main/java/io/harness/perpetualtask/PerpetualTaskScheduleServiceImpl.java:119]: Do we need to delete PerpetualTaskRecord?

File: 400-rest/src/main/java/software/wings/service/impl/CommandServiceImpl.java [hunk 2]
```diff
@@ -15,13 +15,14 @@
...
 import software.wings.service.intfc.CommandService;
+import software.wings.service.intfc.ownership.OwnedByAccount;
 import software.wings.service.intfc.yaml.YamlPushService;
 
 import com.google.inject.Inject;
 import com.google.inject.Singleton;
 
 @Singleton
-public class CommandServiceImpl implements CommandService {
+public class CommandServiceImpl implements CommandService, OwnedByAccount {
```
Comment (abhinav-harness) [inline 400-rest/src/main/java/software/wings/service/impl/CommandServiceImpl.java:25]: add ownedByAccount at Service not at Impl

**Comment by boopesh007** [comment]
*[`400-rest/src/main/java/io/harness/perpetualtask/PerpetualTaskScheduleServiceImpl.java` line 119 at r5](https://reviewable.io/reviews/harness/harness-core/40372#-NH9l538th5UbHlzDWmv-r5-119:-NHh_GpdBc06xiFfNQwQ:bmcmsdv) ([raw file](https://github.com/harness/harness-core/blob/c3e2cb482efc2baeb6e10c7593babfdece237b23/400-rest/src/main/java/io/harness/perpetualtask/PerpetualTaskScheduleServiceImpl.java#L119)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

Do we need to delete PerpetualTaskRecord?
</blockquote></details>

Why shouldn't we? We are deleting only after accounts are churned right.

**Comment by boopesh007** [comment]
*[`400-rest/src/main/java/software/wings/service/impl/CommandServiceImpl.java` line 25 at r5](https://reviewable.io/reviews/harness/harness-core/40372#-NHOW6g8QO5zP_Vy_VJu-r5-25:-NHh__8e5lU6aPMDI4Kh:b2ni9ah) ([raw file](https://github.com/harness/harness-core/blob/c3e2cb482efc2baeb6e10c7593babfdece237b23/400-rest/src/main/java/software/wings/service/impl/CommandServiceImpl.java#L25)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

add ownedByAccount at Service not at Impl
</blockquote></details>

ok

**Comment by abhinav-harness** [comment]
*[`400-rest/src/main/java/io/harness/perpetualtask/PerpetualTaskScheduleServiceImpl.java` line 119 at r5](https://reviewable.io/reviews/harness/harness-core/40372#-NH9l538th5UbHlzDWmv-r5-119:-NHhdXzt9EySUZmOQUTd:bhz28c0) ([raw file](https://github.com/harness/harness-core/blob/c3e2cb482efc2baeb6e10c7593babfdece237b23/400-rest/src/main/java/io/harness/perpetualtask/PerpetualTaskScheduleServiceImpl.java#L119)):*
<details><summary><i>Previously, boopesh007 (Boopesh Shanmugam) wrote…</i></summary><blockquote>

Why shouldn't we? We are deleting only after accounts are churned right.
</blockquote></details>

I mean is perpetual task record deletion taken care?

**Comment by boopesh007** [comment]
*[`400-rest/src/main/java/io/harness/perpetualtask/PerpetualTaskScheduleServiceImpl.java` line 119 at r5](https://reviewable.io/reviews/harness/harness-core/40372#-NH9l538th5UbHlzDWmv-r5-119:-NHhe-nv8N7QLolx-4Np:bssa411) ([raw file](https://github.com/harness/harness-core/blob/c3e2cb482efc2baeb6e10c7593babfdece237b23/400-rest/src/main/java/io/harness/perpetualtask/PerpetualTaskScheduleServiceImpl.java#L119)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

I mean is perpetual task record deletion taken care?
</blockquote></details>

Yes, its a subtype of AccountAccess

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 16 of 16 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40372)* status: 17 file(s) to review. 1 of 2 approvals obtained.

**Review by inderschera — APPROVED** [review]
> Reviewed 1 of 11 files at r1, 3 of 7 files at r2, 7 of 9 files at r3, 1 of 1 files at r5, 12 of 16 files at r6, 7 of 7 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40372)* status: 7 file(s) to review. 1 of 2 approvals obtained.

**Pull request merged**